tests/unit/browser/test_commands.py

```diff
@@ -0,0 +1,212 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Test TabbedBrowser command dispatche."""
+
+from unittest import mock
+
+import pytest
+
+from qutebrowser.browser import commands
+from qutebrowser.completion.models import completionmodel
+from qutebrowser.commands import cmdexc
+from qutebrowser.mainwindow import tabbedbrowser, tabwidget
+from qutebrowser.utils import objreg
+
+
+class TestBuffer:
```

Comment:
Since this is getting quite long - what about moving buffer tests to their own `test_buffer.py` as top-level functions, and adding a `test/unit/browser/commands` subfolder instead?